lib/src/session.rs

```diff
@@ -65,6 +65,10 @@ pub struct Session {
     ///
     /// Populated prior to guest execution, and never modified.
     pub(crate) backends: Arc<Backends>,
+    /// The TLS configuration for this execution.
+    ///
+    /// Populated prior to guest execution, and never modified.
+    pub(crate) tls_config: Arc<rustls::ClientConfig>,
```

Comment:
```suggestion
    tls_config: Arc<rustls::ClientConfig>,
```

because we mention that this is never modified in the comment, perhaps we can remove the `pub(crate)` visibility here, to help enforce that?

that leaves the `tls_config` accessor below as the sole interface through which other modules can acquire a reference to this :slightly_smiling_face:

Reply:
(this is one of those "_take it or leave it_" comments)

Reply:
oh good catch, i was following the pattern of the other read-only items. they should all be using accessors rather than `pub(crate)`, so I'll add a commit to do that :+1: